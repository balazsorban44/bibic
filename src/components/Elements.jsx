import React, {useState, useEffect} from 'react'
import {Link} from 'react-router-dom'
// For the slider elements
import {colors} from '../utils/colors'
import {useTranslation} from 'react-i18next'

export const Loading = ({fullPage, timeout = 10000}) => {

  const [timedOut, setTimedOut] = useState()

  useEffect(() => {
    const tick = () => setTimedOut(true)
    const timeoutId = setTimeout(tick, timeout)
    return () => clearTimeout(timeoutId)
  }, [timeout])

  const [t] = useTranslation()
  if (timedOut) {
    return (
      <p className="not-available">
        {t("content-unavailable")}
      </p>
    )
  }

  const style = fullPage ? {
    width: "100vw",
    height: "100vh",
    display: "grid",
    justifyItems: "center",
    alignItems: "center",
    backgroundColor: colors.light
  } : undefined

  return (
    <div style={style}>
      <div className="loading"><div className="spinner"/></div>
    </div>
  )


}

export const Button = ({label, to}) => <Link {...{to}} >{label}</Link>