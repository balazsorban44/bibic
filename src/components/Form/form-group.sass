@import "ui/_vars"

@media (min-width: 760px)
  .form-group
    margin-top: 1em
    display: grid
    grid-template-columns: 1fr 1fr
    grid-gap: 1em
    background-color: transparentize($accent-color, .95)
    border: 1px solid $accent-color-dark
    border-radius: 12px
    padding: 1em
