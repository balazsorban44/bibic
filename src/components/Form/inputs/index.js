import Send from "./Send"
import PeopleCount from "./PeopleCount"
import Children from "./Children"

export {Send, PeopleCount, Children}