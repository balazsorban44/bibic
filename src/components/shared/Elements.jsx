import React from 'react'

// For the slider elements
import arrow from '../../assets/icons/arrow.svg'
import swipe from '../../assets/icons/swipe.svg'

export const Prev = ({onClick}) => (
  <div
    className="slider-arrow slider-arrow-prev"
    onClick={onClick}
  ><img alt="" src={arrow}/></div>
)

export const Next = ({onClick}) => (
  <div
    className="slider-arrow slider-arrow-next"
    onClick={onClick}
  ><img alt="" src={arrow}/></div>
)

export const SwipeIcon = (props, {isShowingSwipe = true}) => <img {...props} className={`swipe-icon ${!isShowingSwipe ? "is-touched": ""}`} alt="" src={swipe}/>




export const Loading = () => <div className="loading"><div className="spinner"/></div>