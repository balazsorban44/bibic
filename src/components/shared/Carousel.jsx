import React, {Component} from 'react'
import withRouter from 'react-router/withRouter'
import Slide from "react-reveal/Slide"
import makeCarousel from "react-reveal/makeCarousel"
import {withStore} from '../db'
import {Loading} from './Elements'

export class Carousel extends Component {

  componentDidMount() {
    window.scrollTo(0, 0)
  }

  render() {
    let {match: {path}} = this.props
    const {
      title, subtitle, galleries
    } = this.props
    path = path.replace("/", "")
    const children = galleries[path] ?
      Object
        .values(galleries[path])
        .map((itemProps, key) =>
          <Slide
            key={key}
            right
          >
            <CarouselItem
              className="carousel-item"
              {...itemProps}
            />
          </Slide>
        ) :
      <Loading/>
    return(
      <section
        className="carousel-wrapper"
        id={path}
      >
        <div className="title">
          <h2>{title}</h2>
          {subtitle && <h3>{subtitle}</h3> }
        </div>
        <CarouselWrapper defaultWait={10000}>
          {children}
        </CarouselWrapper>
      </section>
    )}
}

const CarouselWrapper = makeCarousel(({
  position, handleClick, children
}) =>
  <div className="carousel">{children}
    <span
      className="carousel-arrow carousel-arrow-prev"
      data-position={position - 1}
      onClick={handleClick}
    ><span className="arrow-icon"/></span>
    <span
      className="carousel-arrow carousel-arrow-next"
      data-position={position + 1}
      onClick={handleClick}
    ><span className="arrow-icon"/></span>
  </div>)

export default withRouter(withStore(Carousel))


export const CarouselItem = ({
  title, desc, SIZE_640, SIZE_1024, SIZE_1440, SIZE_ORIGINAL, className
}) =>
  <div className={className}>
    <span>
      <img
        alt="blurred background"
        className="carousel-item-bg"
        src={SIZE_640}
      />
    </span>
    <div className="carousel-item-picture">
      <picture
        className="carousel-item-img"
      >
        <source
          media="(max-width: 640px)"
          srcSet={SIZE_640}
        />
        <source
          media="(max-width: 1024px)"
          srcSet={SIZE_1024}
        />
        <source
          media="(max-width: 1440px)"
          srcSet={SIZE_1440}
        />
        <img
          alt={title}
          src={SIZE_ORIGINAL}
        />
      </picture>
    </div>
    {desc ?
      <div className="carousel-item-desc">
        <h4>{title}</h4>
        <p>{desc}</p>
      </div> : null
    }
  </div>
