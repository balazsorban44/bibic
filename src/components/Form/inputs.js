import React, { Component, Fragment } from 'react'
import {valid} from '../../utils/validate'
import moment from 'moment'
import {Loading} from "../shared/Elements"
class PersonalDetail extends Component {

  state = {
    type: null,
    placeholder: "",
    label: "",
    name: "",
    value: "",
    hasFootnote: false,
    error: false,
    errorMessage: ""
  }

  componentDidMount() {
    const {type, label, name, placeholder, hasFootnote, errorMessage} = this.props
    this.setState({type, label, name, placeholder, hasFootnote, errorMessage})
  }

  handleChange = ({target: {name, value}}) => this.setState({name, value})

  handleBlur = ({target: {name, value}}) => {
    let error = false
    
    if (valid[name](value)) {
      this.props.onChange(name, value)
    } else {
      this.props.onChange(name, "")
      error = true
    }
    
    this.setState({error})
    // do not fire notification, if the user did not fill in any info, or the info was corrected
    if (error && value !== "") {
      this.props.notification( 
        <p>
          {this.state.errorMessage}<br/>
          Kérjük próbálja újra.
          Ha úgy gondolja hiba történt, kérjük írjon:<br/>
          <a href="mailto:hiba@bibicvedeghazak.hu">hiba@bibicvedeghazak.hu</a>
        </p>, {autoClose: 5000}
      )  
    }
  }

  render() {
    const {type, label, name, placeholder, value, error, hasFootnote} = this.state
    return (
      <div className={`input-box personal-detail ${error ? "input-error" : ""}`}>
        <label className={hasFootnote ? "footnote-asterix" : ""} htmlFor={name}>{label}</label>
        <input {...{name, value, placeholder}}
          type={type || "text"}
          onChange={this.handleChange}
          onBlur={this.handleBlur}
        />
      </div>
    )
  }
}




const Date = ({onClick, name, label, value, hasFootnote}) => (
  <div 
    className="input-box"
    onClick={() => onClick("Kérjük válasszon érkezési és távozási dátumot a naptáron!")} 
  >
    <label 
      className={hasFootnote ? "footnote-asterix" : ""}
      htmlFor={name}
    >
      {label}
    </label>
    <input readOnly {...{name}} value={moment(value).format("LL")}/>
  </div>
)


class PeopleCount extends Component {

  state = {
    hasFootnote: false,
    min: 0,
    max: 0,
    name: "",
    label: "",
    placeholder: "",
    value: 0
  }

  componentDidMount() {
    const {hasFootnote, min, max, name, label, placeholder, value} = this.props
    this.setState({
      value: parseInt(value, 10),
      hasFootnote, min, max, name, label, placeholder
    })
  }
  
  componentWillReceiveProps = ({value, max}) => {
    this.setState({
      value: parseInt(value, 10), max
    })
  }

  handleChange = e => {
    e.preventDefault()
    let {target: {name, value}} = e
    value = parseInt(value, 10)
    const {min, max} = this.state
    if (value < min) {
      value = min
    } else if (value > max) {
      value = max
    }
    
    this.setState({value})
    this.props.onChange(name, value)
  }

  render() {
    const {hasFootnote, min, max, name, label, placeholder, value} = this.state
    return (
      <div className="input-box people-count">
        <label 
          className={hasFootnote ? "footnote-asterix": ""}
          htmlFor={name}
        >
          {label}
        </label>
        <input {...{min, max, placeholder, name}}
          type="number"
          value={value + ""}
          readOnly
        />
        <div className="people-buttons">
          <button {...{name}}
            value={value - 1}
            onClick={this.handleChange}
          >-</button>
          <button {...{name}}
            value={value + 1}
            onClick={this.handleChange}
          >+</button>
        </div>
      </div>
    )
  }
}



class Children extends Component {

  state = {
    values: [],
    max: 100
  }

  componentWillReceiveProps = ({values, max}) => this.setState({values, max})
  
  componentDidMount() {
    const {values, max} = this.props
    this.setState({values, max})
  }


  handleChange = (name, value) => {
    let  {values, max} = this.state
    value = parseInt(value, 10)
    
    if (values.length < value) {
      values.push("6-12")
      values = values.slice(0, max)
    }
    if (values.length > value) {
      values.pop()
    }
    
    this.props.onChange("children", values)
  }

  handleSelect = ({target: {name, value}}) => {
    let {values} = this.state
    values[name] = value
    this.setState({
      values
    })
    this.props.onChange("children", values)
  }

  render() {
    const {hasFootnote, name, label, max} = this.props
    const {values} = this.state
    
    return (
      <Fragment>
        <PeopleCount {...{hasFootnote, name, label, max}}
          min={0} placeholder={0}
          onChange={this.handleChange}
          value={values.length}
        />
        <div className="children-age">
          {values.length >= 1 && values.map((childAge, index) => {
            return (
              <div key={index} className="input-box child-age">
                <label htmlFor={index}>{index+1}. kor</label>
                <select 
                  value={childAge}
                  name={index}
                  onChange={this.handleSelect}
                >
                  <option value="0-6" >0-6 év</option>
                  <option value="6-12">6-12 év</option>
                </select>
              </div>
            )
          })}
        </div>
      </Fragment>
    )
  }
}


const Service = ({label, name, onChange, checked, value}) => (
  <div className="input-box service">
    <label htmlFor={value}>{label}</label>
    <input {...{name, value, checked}}
      type="radio"
      onChange={({target: {name, value}}) => onChange(name, value)}
      id={value}
    />
  </div>
)


const Send = ({disabled, onClick, children}) =>
  <Fragment>
    {disabled ?
      <div style={{padding: "24px 0 48px"}} >
        <Loading/>
      </div> :
      <button
        {...{onClick}} 
        type="submit"
        className={`submit-reservation ${disabled ? "active-reserving": ""}`}
      >{children}
      </button>
    }
  </Fragment>


export {PersonalDetail, Children, Date, PeopleCount, Service, Send}