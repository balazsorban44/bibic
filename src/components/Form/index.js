import React from "react"
import {Send, PeopleCount, Children} from "./inputs"


const FormSection = ({title, children}) =>
  <>
    <h4>{title}</h4>
    {children}
  </>


const FormGroup = ({title, className, children, footnote}) =>
  <>
    {title && <h5>{title}</h5>}
    <div className={`form-group ${className}`}>
      {children}
      {footnote &&<span className="footnote">{footnote}</span>}
    </div>
  </>


export {FormSection, FormGroup, Send, PeopleCount, Children}