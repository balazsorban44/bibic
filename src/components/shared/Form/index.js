import React from 'react'
import {Date, Send, PeopleCount, Children, Service} from "./inputs"


const FormSection = ({title, children}) =>
  <>
    <h4>{title}</h4>
    {children}
  </>


const FormGroup = ({title, className, children, footnote}) =>
  <>
    {title && <h5>{title}</h5>}
    <div className={`form-group ${className}`}>
      {children}
      {footnote &&<span className="footnote">{footnote}</span>}
    </div>
  </>


export {FormSection, FormGroup, Date, Send, PeopleCount, Children, Service}