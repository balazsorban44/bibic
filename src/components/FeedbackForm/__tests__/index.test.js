import {
  FeedbackForm, RatingSlider, RatingValue, FeedbackDone
} from "../"
import {Loading} from "../../shared/Elements"
import {Send} from "../../shared/Form"

global.scrollTo = jest.fn()

describe("FeedbackForm component", () => {
  const props = {submitFeedback: jest.fn().mockResolvedValue({error: true, success: true})}
  let wrapper = shallow(<FeedbackForm {...props}/>)


  beforeEach(() => {
    wrapper.setState({error: false, success: false})
    wrapper.setProps({loading: false})
  })

  it("renders correctly", () => {
    expect(wrapper).toHaveLength(1)
  })

  it("scrolled to top", () => {
    expect(scrollTo).toBeCalledWith(0, 0)
  })

  it("Loading shows", () => {
    wrapper.setProps({loading: true})
    expect(wrapper.find(Loading).length).toBe(1)
  })

  it("Success shows", () => {
    wrapper.setState({success: true})
    expect(wrapper.find(FeedbackDone).length).toBe(1)
  })

  it("Error shows", () => {
    wrapper.setState({error: true})
    expect(wrapper.find(FeedbackDone).length).toBe(1)
  })

  it("submitting invoked", () => {
    wrapper.find(Send).simulate("click", {preventDefault: jest.fn()})
    expect(props.submitFeedback).toBeCalledWith({
      content: "",
      id: null,
      ratings: {
        cleanliness: 0,
        coffee: 0,
        comfort: 0,
        food: 0,
        services: 0,
        staff: 0
      }
    })
    // expect(wrapper.state("success")).toBe(true)
    // expect(wrapper.state("error")).toBe(true)
  })

  describe("one-click submitting", () => {
    it("rating is given", () => {
      wrapper = shallow(
        <FeedbackForm
          {...{...props, location: {search: "?id=id&one-click=1&rating=5"}}}
        />
      )
      expect(props.submitFeedback).toBeCalledWith({id: "id",
        ratings: {
          cleanliness: 5,
          coffee: 5,
          comfort: 5,
          food: 5,
          services: 5,
          staff: 5
        }})
    })
    it("default rating is 0", () => {
      wrapper = shallow(
        <FeedbackForm
          {...{...props, location: {search: "?id=id&one-click=1"}}}
        />
      )
      expect(props.submitFeedback).toBeCalledWith({id: "id",
        ratings: {
          cleanliness: 0,
          coffee: 0,
          comfort: 0,
          food: 0,
          services: 0,
          staff: 0
        }})
    })
  })

  describe("show full form (eg.: NO one-click submitting)", () => {
    it("set id in the state", () => {
      wrapper = shallow(
        <FeedbackForm
          {...{...props, location: {search: "?id=id"}}}
        />
      )
      expect(wrapper.state("feedback").id).toBe("id")
    })
  })

  describe("change correctly", () => {
    const name = "name"
    const value = "value"
    it("textarea", () => {
      wrapper.find("textarea").simulate("change", {target: {name, value}})
      expect(wrapper.state("feedback")[name]).toBe(value)
    })

    it("Rating slider click", () => {
      wrapper.find(RatingSlider).first().simulate("change", name, value)
      expect(wrapper.state("feedback").ratings[name]).toBe(value)
    })
  })

})


describe("RatingSlider component", () => {
  const wrapper = shallow(<RatingSlider/>)

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })

})


describe("RatingValue component", () => {
  const props = {
    onChange: jest.fn(), name: "name", value: "value"
  }
  const wrapper = shallow(<RatingValue {...props}/>)

  it("renders correctly", () => {
    expect(wrapper).toHaveLength(1)
  })

  it("click propagated", () => {
    wrapper.simulate("click")
    expect(props.onChange).toBeCalledWith(props.name, props.value)
  })
  it("gets active class", () => {
    wrapper.setProps({isActive: true})
    expect(wrapper.hasClass("rating-value-active")).toBe(true)
  })
})

describe("FeedbackDone component", () => {
  const wrapper = shallow(<FeedbackDone/>)
  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })
})