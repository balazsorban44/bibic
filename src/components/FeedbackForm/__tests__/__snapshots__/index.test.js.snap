// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackDone component renders correctly 1`] = `ShallowWrapper {}`;

exports[`RatingSlider component renders correctly 1`] = `ShallowWrapper {}`;
